contracts/Orchestrator.sol

```diff
@@ -90,7 +90,6 @@ contract Orchestrator is Ownable {
             transactions[index] = transactions[transactions.length - 1];
         }
 
-        delete transactions[transactions.length - 1];
```

Comment:
That's hidden and good to know! we have this delete pattern in multiple places in our codebase.